### DISCUSSION TITLE
Support/plugins/extensions for editing text based file in browser?

### DISCUSSION THREAD

**Discussion by shubhendumadhukar** [OpusCapita/filemanager#324] [Q&A]
I have been using filemanager for one of my projects and it has been going great. I have some text based files on a remote server which I am able to access using filemanager in my browser. However, in order to edit the contents of the file, I have to download it, modify it and re-upload it.

I was wondering, if there are any available extensions/plugins that can be used with filemanager to facilitate this. Theoretically it seems simple enough to implement it. Editing a file can occur within a codemirror textarea inside a modal. I am not sure how the backend would work, but I do see great value in it. Thoughts?

**Comment by estambakio-sc** [comment]
In theory it can be done in a way that upon "edit" button click 
- a file is downloaded (a byte stream is fetched into a variable on client side)
- editor of any kind is opened, downloaded file is passed as input to it 
- when editor is finished then in a callback "upload" capability of connector can be called to upload changed file again 

Filemanager `client-react` doesn't have interface for plug-in components like this at the moment, therefore such editor should be developed in a fork I guess.
